packages/hardhat-core/test/internal/artifacts.ts

```diff
@@ -1309,6 +1309,40 @@ describe("Artifacts class", function () {
       });
     });
   });
+
+  describe("artifactExists", function () {
+    useTmpDir("artifacts");
+
+    it("should check that the artifact exists", async function () {
+      const artifact = getArtifactFromContractOutput(
+        "source.sol",
+        "Interface",
+        {
+          ...COMPILER_OUTPUTS.Interface,
+          evm: undefined,
+        }
+      );
+
+      const artifacts = new Artifacts(this.tmpDir);
+      await artifacts.saveArtifactAndDebugFile(artifact, "");
+
+      // FQN
+      assert.isTrue(await artifacts.artifactExists("source.sol:Interface"));
+      // non-FQN
+      assert.isTrue(await artifacts.artifactExists("Interface"));
+    });
+
+    it("should check that the artifact doesn't exist", async function () {
+      const artifacts = new Artifacts(this.tmpDir);
+
+      // FQN, missing artifact
+      assert.isFalse(await artifacts.artifactExists("invalid/source.sol:A"));
```

Comment:
This assertion doesn't match the comment, does it? Maybe you meant this?

```suggestion
      // FQN, missing file
      assert.isFalse(await artifacts.artifactExists("invalid/source.sol:Interface"));
      // FQN, missing artifact
      assert.isFalse(await artifacts.artifactExists("source.sol:A"));
```

Reply:
That's a valid difference, but for it to be valid we'd need to generate an artifact like we do in the successful case. And it actually makes sense to add it for the rest of the assertions as well, as they don't make too much sense otherwise. Thanks!